__tests__/api.test.js

```diff
@@ -161,6 +161,56 @@ describe("GET /api/articles", () => {
             })
         })
     })
+    test("Checking that the articles are sorted in descending order based on the date", () => {
+        
+        return request(app).get("/api/articles").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("created_at", {descending : true});
+        })
+    })
+    test("200 - Adding 'sort_by' inside the query", () => {
+        return request(app).get("/api/articles?sort_by=comment_count").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("comment_count", {descending: true, coerce: true})
+        })
+    })
+    test("200 - Adding 'order' inside the query", () => {
+        return request(app).get("/api/articles?sort_by=votes&order=ASC").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toBeSortedBy("votes");
+        })
+    })
+    test("200 - Adding 'topic' inside the query", () => {
+        return request(app).get("/api/articles?topic=cats").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toHaveLength(1);
+            articles.forEach((article) => {
+                expect(article.topic).toBe("cats");
+            })
+        })
+    })
+    test("200 - Testing complete query", () => {
+        return request(app).get("/api/articles?topic=mitch&sort_by=votes&order=ASC").expect(200).then((res) => {
+            const {articles} = res.body;
+            expect(articles).toHaveLength(11);
+            expect(articles).toBeSortedBy("votes");
+            articles.forEach((article) => {
+                expect(article.topic).toBe("mitch");
+            })
+        })
+    })
+    test("400 - Setting invalid sort_by value", () => {
+        return request(app).get("/api/articles?sort_by=apples").expect(400).then((res) => {
+            const {message} = res.body;
+            expect(message).toBe("Invalid sort_by value");
+        })
+    })
+    test("404 - Setting invalid topic value", () => {
+        return request(app).get("/api/articles?topic=apples").expect(404).then((res) => {
+            const {message} = res.body;
+            expect(message).toBe("No articles found");
+        })
+    })
```

Comment:
> Have you tested invalid order query?

well the way i set the order query up it will either be ASC or DESC, if the user inputs ASC then it will be added, else it will default to DESC and ignore any other input